x-pack/plugins/apm/server/lib/transaction_groups/transform.test.ts

```diff
@@ -54,4 +55,27 @@ describe('transactionGroupsTransformer', () => {
       }
     ]);
   });
+
+  it('should calculate impact from sum', () => {
+    const getBucket = (sum: number) => ({
+      key: 'POST /api/orders',
+      doc_count: 180,
+      avg: { value: 300000 },
+      p95: { values: { '95.0': 320000 } },
+      sum: { value: sum },
+      sample: { hits: { total: 180, hits: [{ _source: 'sample source' }] } }
+    });
+
+    const response = ({
+      aggregations: {
+        transactions: { buckets: [getBucket(10), getBucket(20), getBucket(50)] }
+      }
+    } as unknown) as ESResponse;
```

Comment:
why is `as unknown` necessary when it get interpreted `as ESResponse` later on?

Reply:
Without first casting it to `unknown`, it complains because the object contains properties that are not defined on `ESResponse`. I think this is due to `@types/elasticsearch` not having everything defined properly.

I wish we didn't have to do this, because it's definitely defeating the purpose of a type system, but I think it's the most pragmatic solution for now.

Reply:
This is the tsc error I get without unknown:

```
Conversion of type '{ took: number; timed_out: false; _shards: { total: number; successful: number; skipped: number; failed: number; }; hits: { total: number; max_score: null; hits: never[]; }; aggregations: { transactions: { ...; }; }; }' to type 'AggregationSearchResponse<void, Aggs>' may be a mistake because neither type sufficiently overlaps with the other. If this was intentional, convert the expression to 'unknown' first.
  Types of property 'aggregations' are incompatible.
    Type '{ transactions: { doc_count_error_upper_bound: number; sum_other_doc_count: number; buckets: ({ key: string; doc_count: number; avg: { value: number; }; p95: { values: { '95.0': number; }; }; sample: { hits: { ...; }; }; } | ... 12 more ... | { ...; })[]; }; }' is not comparable to type 'Aggs'.
      Types of property 'transactions' are incompatible.
        Type '{ doc_count_error_upper_bound: number; sum_other_doc_count: number; buckets: ({ key: string; doc_count: number; avg: { value: number; }; p95: { values: { '95.0': number; }; }; sample: { hits: { total: number; max_score: null; hits: { ...; }[]; }; }; } | ... 12 more ... | { ...; })[]; }' is not comparable to type '{ buckets: Bucket[]; }'.
          Types of property 'buckets' are incompatible.
            Type '({ key: string; doc_count: number; avg: { value: number; }; p95: { values: { '95.0': number; }; }; sample: { hits: { total: number; max_score: null; hits: { _index: string; _type: string; _id: string; _score: null; _source: { ...; }; sort: number[]; }[]; }; }; } | ... 12 more ... | { ...; })[]' is not comparable to type 'Bucket[]'.
              Type '{ key: string; doc_count: number; avg: { value: number; }; p95: { values: { '95.0': number; }; }; sample: { hits: { total: number; max_score: null; hits: { _index: string; _type: string; _id: string; _score: null; _source: { ...; }; sort: number[]; }[]; }; }; } | ... 12 more ... | { ...; }' is not comparable to type 'Bucket'.
                Type '{ key: string; doc_count: number; avg: { value: number; }; p95: { values: { '95.0': number; }; }; sample: { hits: { total: number; max_score: null; hits: { _index: string; _type: string; _id: string; _score: null; _source: { ...; }; sort: number[]; }[]; }; }; }' is not comparable to type 'Bucket'.
                  Types of property 'sample' are incompatible.
                    Type '{ hits: { total: number; max_score: null; hits: { _index: string; _type: string; _id: string; _score: null; _source: { '@timestamp': string; processor: { name: "transaction"; event: "transaction"; }; transaction: { ...; }; ... 4 more ...; host: { ...; }; }; sort: number[]; }[]; }; }' is not comparable to type '{ hits: { total: number; max_score: number | null; hits: { _source: Transaction; }[]; }; }'.
                      Types of property 'hits' are incompatible.
                        Type '{ total: number; max_score: null; hits: { _index: string; _type: string; _id: string; _score: null; _source: { '@timestamp': string; processor: { name: "transaction"; event: "transaction"; }; transaction: { ...; }; ... 4 more ...; host: { ...; }; }; sort: number[]; }[]; }' is not comparable to type '{ total: number; max_score: number | null; hits: { _source: Transaction; }[]; }'.
                          Types of property 'hits' are incompatible.
                            Type '{ _index: string; _type: string; _id: string; _score: null; _source: { '@timestamp': string; processor: { name: "transaction"; event: "transaction"; }; transaction: { name: string; duration: { us: number; }; ... 4 more ...; id: string; }; ... 4 more ...; host: { ...; }; }; sort: number[]; }[]' is not comparable to type '{ _source: Transaction; }[]'.
                              Type '{ _index: string; _type: string; _id: string; _score: null; _source: { '@timestamp': string; processor: { name: "transaction"; event: "transaction"; }; transaction: { name: string; duration: { us: number; }; ... 4 more ...; id: string; }; ... 4 more ...; host: { ...; }; }; sort: number[]; }' is not comparable to type '{ _source: Transaction; }'.
                                Object literal may only specify known properties, and '_index' does not exist in type '{ _source: Transaction; }'. [2352]
(property) _index: string
```